### DISCUSSION TITLE
Parsing string values as JSON-strings bug?

### DISCUSSION THREAD

**Discussion by allanbbs** [nlohmann/json#4397] [Q&A] [answered]
Hi.
Suppose I want to have the following string:
{
"dummy" : "\\"xd\\""
}

If I use the following snippet:
`
const auto value = json::parse("\"{\"dummy\":\"\"xd\"\"}\"");
`
it throws an exception. Is this supposed to hapen? Should I need to escape the inner slashes/quotes as well?

**Comment by gregmarr** [comment]
> Suppose I want to have the following string:
```
{
"dummy" : ""xd""
}
```
That's not valid JSON.  You want this if you want the value of `dummy` to be a string with the content `"xd"`:
```
{
"dummy" : "\"xd\""
}
```

**Reply by allanbbs** [reply]
I did not escape the quotes in my question. Fixed, thanks

**Reply by gregmarr** [reply, marked as answer]
When converting that JSON string to a C++ string, you need to escape both `"` and `\`, or use a raw string literal.

```
json.parse(R"({
"dummy" : "\"xd\""
})");
```
https://www.godbolt.org/z/q6b3nv4Md

**Reply by allanbbs** [reply]
Thanks!
